### DISCUSSION TITLE
Use id in tar_render_rep() to name output reports?

### DISCUSSION THREAD

**Discussion by gorkang** [ropensci/tarchetypes#65] [Help] [answered]
I am creating personalized pdf reports for the participants in a study using the amazing `tar_render_rep()`. 

Using params, I pass the `id` of the participants (e.g. `id = abc`) to the Rmarkdown, so I can show the results for each participant.

The thing I am struggling with is **how to use the same `id` to name the output reports**. Right now, the reports are named `report_Participants_5cd8dc6f.pdf` and similar, but I would need them to be `report_Participants_[id].pdf`, for example: `report_Participants_abc.pdf`

This is the code I am using:  
```
tar_render_rep(report_Participants, 
               path = here::here("doc/report_Participants.Rmd"), 
               params = tibble::tibble(id = c(DF_analysis %>% pull(id))),
               output_dir = paste0("outputs/reports/personalized_reports"))
```

Any help would be appreciated. 

Thanks!

**Comment by wlandau** [comment, marked as answer]
You can specify an `output_file` column of `params` to manually control the base name of each output file: https://github.com/ropensci/tarchetypes/blob/30b155741ac7f28ab3b67f123475b10fcf60e679/R/tar_render_rep_raw.R#L48-L52

**Reply by gorkang** [reply]
Thanks a lot, and sorry I missed that in the help file.
